### PR TITLE
Isolate large kubemark executors to heavy machines

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/31589

@gmarek

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/478)

### CODE REVIEW

File: jenkins/attach_agent.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 SHARED = False
 
 INFO = {
-    'heavy': ('build unittest', 1, EXCLUSIVE),
+    'heavy': ('build unittest scalability', 1, EXCLUSIVE),
```
Comment (fejta) [inline jenkins/attach_agent.py]: Let's not do this until we also update agent-ctl.sh and/or attach_agent.py to be able to modify the set of tags on existing agents.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-kubemark.yaml [hunk 2]
```diff
@@ -196,6 +204,9 @@
                 export KUBE_OS_DISTRIBUTION="gci"
         - 'gce-scale':
             description: 'Run Density test on Kubemark in very large cluster. Currently only scheduled to run every 12 hours so as not to waste too many resources.'
+            # TODO: We should replace it with 'scalability' instead of using
```
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-kubemark.yaml]: Please note that most agents will not have this tag.

**Comment by fejta** [comment]
The change to `build` LGTM, but we don't have a great way now to update agent labels without deleting all the nodes, which takes forever.

File: jenkins/attach_agent.py (hunk 1 above)
Comment (wojtek-t) [inline jenkins/attach_agent.py]: Added a TODO instead.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-kubemark.yaml (hunk 2 above)
Comment (wojtek-t) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-kubemark.yaml]: Clarified the comment.

**Comment by wojtek-t** [comment]
@fejta - I applied your comments. 

Thanks for the review.

**Comment by fejta** [comment]
FYI @ixdy

**Comment by ixdy** [comment]
I'm not entirely convinced the Jenkins agents were overloaded, but I guess we can collect data now and see.

**Comment by wojtek-t** [comment]
>
I'm not entirely convinced the Jenkins agents were overloaded, but I guess we can collect data now and see.

@ixdy - I'm 99% sure this is overloaded Jenkins - see this as an example:
https://github.com/kubernetes/kubernetes/issues/31589#issuecomment-243350367

**Comment by ixdy** [comment]
should we revert this, given that #491 was determined to be the actual fix?

**Comment by wojtek-t** [comment]
We can try reverting.

**Pull request merged**